python/examples/atomistic/3-atomistic-model-with-nl.py

```diff
@@ -0,0 +1,300 @@
+            sigma_r_12 = sigma_r_6 * sigma_r_6
+            e = 4.0 * self._epsilon * (sigma_r_12 - sigma_r_6) - self._shift
+
+            # We only compute each pair once (full_list=False in self._nl_options),
+            # and assign half of the energy to each atom
+            energy = energy.index_add(0, all_i, e, alpha=0.5)
+            energy = energy.index_add(0, all_j, e, alpha=0.5)
+
+            all_energies.append(energy.sum(0, keepdim=True))
+
+        samples = Labels(
+            ["system"], torch.arange(len(systems), device=device).reshape(-1, 1)
+        )
+
+        block = TensorBlock(
+            values=torch.vstack(all_energies).reshape(-1, 1),
+            samples=samples,
+            components=torch.jit.annotate(List[Labels], []),
+            properties=Labels(["energy"], torch.tensor([[0]], device=device)),
+        )
+        return {
+            "energy": TensorMap(
+                Labels("_", torch.tensor([[0]], device=device)), [block]
+            ),
+        }
+
+
+# %%
+#
+# In the model above, in addition to the required ``__init__()`` and
+# :py:meth:`ModelInterface.forward` methods, we also implemented the
+# :py:meth:`ModelInterface.requested_neighbor_lists` method, which declares the neighbor
+# list our model requires.
+#
+# Load simulation state
+# ------------------------
+#
+# Now that we have a model for the energy of our system, we load a pre-equilibrated
+# :download:`liquid argon system <liquid-argon.xyz>` from a file. The system was
+# generated based on a 5x5x5 supercell of a simple cubic cell with a lattice constant of
+# a = 3.641 Å. After initialization of the velocities, the system was run for 100 ps
+# with the same parameters we will use below.
+
+atoms = ase.io.read("liquid-argon.xyz")
+
+# %%
+#
+# We use a simple cubic (sc) unit cell with a lattice constant of 3.64 Å which should
+# give a density of roughly 1.374 :math:`\mathrm{gcm^{-3}}`.
+
+ax = ase.visualize.plot.plot_atoms(atoms, radii=0.5)
+ax.set_xlabel("$\\AA$")
+ax.set_ylabel("$\\AA$")
+plt.show()
+
+# %%
+#
+# The system already has velocities and the expected density of 1.374 g/cm³.
+
+u_to_g = 1.66053906660e-24
+Å_to_cm = 1e-08
+
+mass_density = sum(atoms.get_masses()) / atoms.cell.volume * u_to_g / Å_to_cm**3
+
+print(f"ρ_m = {mass_density:.3f} g/cm³")
+
+# %%
+#
+# Running the simulation
+# ----------------------
+#
+# The final step to run the simulation is to register our model as the energy calculator
+# for these ``atoms``. This is the job of
+# :py:class:`ase_calculator.MetatensorCalculator`, which takes either an instance of
+# :py:class:`MetatensorAtomisticModel` or the path to a pre-exported model, and allows
+# it to compute the energy, forces, and stress acting on a system.
+#
+# We now define and wrap the model, using the initial positions and the Lennard-Jones
+# parameters taken from `Méndez-Bermúdez et.al, Phys. Commun. 2022
+# <https://dx.doi.org/10.1088/2399-6528/ac62a2>`_.
```

Comment:
we could add a small sentence on units & making sure to declare the right ones in the capabilities